wagtail/admin/utils.py

```diff
@@ -70,3 +77,118 @@ def set_query_params(url: str, params: dict):
     querydict = {key: value for key, value in querydict.items() if value is not None}
     query = urlencode(querydict, doseq=True)
     return urlunsplit((scheme, netloc, path, query, fragment))
+
+
+class TabbedEditHandlerGeneratorMixin:
+    # A class (or import path to a class) that `make_edit_handler()` should
+    # use to generate an edit handler for the model.
+    edit_handler_class = "wagtail.admin.panels.TabbedInterface"
+
+    # A class (or import path for a class) that `make_edit_handler()` should
+    # use to generate individual tabs for the model's edit handler.
+    edit_handler_tab_class = "wagtail.admin.panels.ObjectList"
+
+    # A list of tuples detailing the `name`, `heading` and `classname` for
+    # each tab to be generated by `make_edit_handler()`.
+    edit_handler_tabs = []
```

Comment:
TODO: scale back to just name and heading tuplrs here, and support enums and other iterables